controllers/pipeline/pipeline_adapter.go

```diff
@@ -280,43 +274,17 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (reconciler.OperationRe
 			return reconciler.RequeueWithError(err)
 		}
 
-		dt, err := a.loader.GetDeploymentTargetForDeploymentTargetClaim(a.client, a.context, dtc)
-		if err != nil || dt == nil {
-			a.logger.Error(err, "Failed to find deploymentTarget defined in deploymentTargetClaim", "deploymentTargetClaim.Name", dtc.Name)
-			return reconciler.RequeueWithError(err)
-		}
-
 		binding, err := a.loader.FindExistingSnapshotEnvironmentBinding(a.client, a.context, a.application, testEnvironment)
 		if err != nil || binding == nil {
 			a.logger.Error(err, "Failed to find snapshotEnvironmentBinding associated with environment", "environment.Name", testEnvironment.Name)
 			return reconciler.RequeueWithError(err)
 		}
 
-		a.logger.Info("Deleting deploymentTarget", "deploymentTarget.Name", dt.Name)
-		err = a.client.Delete(a.context, dt)
-		if err != nil {
-			a.logger.Error(err, "Failed to delete the deploymentTarget")
-			return reconciler.RequeueWithError(err)
-		}
-		a.logger.LogAuditEvent("DeploymentTarget deleted", dt, h.LogActionDelete)
-
-		a.logger.Info("Deleting deploymentTargetClaim", "deploymentTargetClaim.Name", dtc.Name)
-		err = a.client.Delete(a.context, dtc)
+		err = h.CleanUpEphemeralEnvironments(a.client, &a.logger, a.context, testEnvironment, dtc)
```

Comment:
Mayber [these code calling Get* functions](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/pipeline/pipeline_adapter.go#L277-L293) can be moved to these  `CleanUpEphemeralEnvironments()` together, then every time they can be run with the deleting code together.

Reply:
I think that the issue with moving those functions is that `CleanUpEphemeralEnvironments()` will also be called as part of the snapshotEnvironmentBinding adapter (#228) , and there will be no need to run [findExistingSnapshotEnvironmentBinding()](https://github.com/redhat-appstudio/integration-service/blob/0046c7466e6dbd07b6f653c452a07c68c840a9e0/controllers/pipeline/pipeline_adapter.go#L289) there since it's already known. 

Additionally,  @Josh-Everett fyi - we can probably remove this call to [GetDeploymentTargetForDeploymentTargetClaim()](https://github.com/redhat-appstudio/integration-service/blob/0046c7466e6dbd07b6f653c452a07c68c840a9e0/controllers/pipeline/pipeline_adapter.go#L283) since we don't need to clean that up any more.

Reply:
@dirgim I might be mistaken but I think the call to `GetDeploymentTargetForDeploymentTargetClaim()` you're referencing should be removed in commit [8e00861](https://github.com/redhat-appstudio/integration-service/commit/8e00861747d591cc96561e35a940cff21ff773bc#diff-b836834f01fdc96e8d8fdf14e0266e2652f2bba56785f2f8dfba25b9b995cf6dL283)

Reply:
It IS however still called within the[ adapter_test](https://github.com/redhat-appstudio/integration-service/blob/8e00861747d591cc96561e35a940cff21ff773bc/controllers/pipeline/pipeline_adapter_test.go#L1404)

Reply:
You're right, this should be good to go now in that case.